src/Builder.php

```diff
@@ -10,278 +10,71 @@
+    public function canOnlyBeUsedAfter(int $notBefore, bool $addHeader = false): Builder;
 
     /**
      * Configures the subject
-     *
-     * @param string $subject
-     * @param bool $addHeader
-     *
-     * @return Builder
      */
-    public function relatedTo(string $subject, bool $addHeader = false): Builder
-    {
-        return $this->setRegisteredClaim('sub', $subject, $addHeader);
-    }
-
-    /**
-     * Configures a registed claim
-     *
-     * @param string $name
-     * @param mixed $value
-     * @param bool $addHeader
-     *
-     * @return Builder
-     */
-    protected function setRegisteredClaim(string $name, $value, bool $addHeader): Builder
-    {
-        $this->with($name, $value);
-
-        if ($addHeader) {
-            $this->headers[$name] = $this->claims[$name];
-        }
-
-        return $this;
-    }
+    public function relatedTo(string $subject, bool $addHeader = false): Builder;
 
     /**
      * Configures a header item
      *
-     * @param string $name
-     * @param mixed $value
-     *
-     * @return Builder
-     *
      * @throws BadMethodCallException When data has been already signed
      */
-    public function withHeader(string $name, $value): Builder
-    {
-        if ($this->signature) {
-            throw new BadMethodCallException('You must unsign before make changes');
-        }
-
-        $this->headers[$name] = $this->claimFactory->create($name, $value);
-
-        return $this;
-    }
+    public function withHeader(string $name, $value): Builder;
 
     /**
      * Configures a claim item
      *
-     * @param string $name
-     * @param mixed $value
-     *
-     * @return Builder
-     *
      * @throws BadMethodCallException When data has been already signed
      */
-    public function with(string $name, $value): Builder
-    {
-        if ($this->signature) {
-            throw new BadMethodCallException('You must unsign before making changes');
-        }
-
-        $this->claims[$name] = $this->claimFactory->create($name, $value);
-
-        return $this;
-    }
-
-    /**
-     * Signs the data
-     *
-     * @param Signer $signer
-     * @param Key $key
-     *
-     * @return Builder
-     */
-    public function sign(Signer $signer, Key $key): Builder
-    {
-        $signer->modifyHeader($this->headers);
-
-        $this->signature = $signer->sign(
-            $this->getToken()->getPayload(),
-            $key
-        );
-
-        return $this;
-    }
+    public function with(string $name, $value): Builder;
 
     /**
-     * Removes the signature from the builder
-     *
-     * @return Builder
+     * Returns a signed token to be used
      */
-    public function unsign(): Builder
-    {
-        $this->signature = null;
-
-        return $this;
-    }
+    public function getToken(Signer $signer, Key $key): Plain;
 
     /**
-     * Returns the resultant token
-     *
-     * @return Token
+     * Returns an unsecured token (not recommended)
```

Comment:
But it can be done in a different PR... I really want to have this merged instead of adding things to the stack